base/applications/network/telnet/src/tnerror.cpp

```
@@ -97,13 +97,37 @@ int wprintit(LPCWSTR it)
     }
     else
     {
+        // calculate the number of bytes needed to store the UTF-8 string
+        int cbMultibyte = WideCharToMultiByte(GetACP(), 0, it, -1, NULL, 0, NULL, NULL);
+        if (cbMultibyte == 0)
+            return 0;
+        if (cbMultibyte < 0)
+            return -1;
+        
+        // allocate the buffer for the UTF-8 string
+        char* szBuffer = new char[cbMultibyte];
+        if (szBuffer == nullptr)
```

Does new[] throw or return NULL?

---

Looking at the rest of telnet's code, there doesn't seem to be any check for NULL after any `new` allocation.

---

As far as I could see, no trace of `try ... except` at all in its code either.
So it seems that it's doing the Unix way of "exception handling" i.e. crashing the program.

---

It does indeed throw... should I add a try-catch here then, or just let the exception propagate to terminate the program?